shared/src/metabase/shared/models/visualization_settings.cljc

```diff
@@ -0,0 +1,410 @@
+  `cljs.core.clj->js` in conjunction with `cljs.js`."
+  [obj]
+  #?(:clj  (json/encode obj)
+     :cljs (.stringify js/JSON (clj->js obj))))
+
+(defn parse-column-ref
+  "Opposite of the `column-ref-for-*` fns. Converts the given string representation of a column settings key into its
+  normalized form. The `col-ref` argument can be a string or keyword (which is produced by YAML parsing, for instance).
+  In case it is a keyword, it will be converted to its full name. \"Full\" means that the portions before and after any
+  slashes will be included verbatim (via the `keyname` helper fn). This is necessary because our serialization code
+  considers a forward slash to be a legitimate portion of a fully qualified name, whereas Clojure considers it to be a
+  namespace/local name separator."
+  {:added "0.40.0"}
+  [col-ref]
+  (let [parts           ((comp vec parse-json-string) (if (keyword? col-ref) (keyname col-ref) col-ref))
+        [kind & params] parts]
+    (case kind
+      "ref"  (let [[[t id & _]] params]
+               (case t "field" (cond (int? id)    {::field-id id}
+                                     (string? id) {::field-qualified-name id})))
+      "name" {::column-name (first params)})))
+
+(defn- with-col-settings [settings]
+  (if (contains? settings ::column-settings)
+    settings
+    (assoc settings ::column-settings {})))
+
+(defn crossfilter-click-action
+  "Creates a crossfilter click action with the given `param-mapping`, in the normalized form."
+  {:added "0.40.0"}
+  [param-mapping]
+  {::click-behavior-type ::cross-filter
+   ::parameter-mapping   param-mapping})
+
+(defn url-click-action
+  "Creates a URL click action linking to a `url-template`, in the normalized form."
+  {:added "0.40.0"}
+  [url-template]
+  {::click-behavior-type ::link
+   ::link-type           ::url
+   ::link-template       url-template})
+
+(defn entity-click-action
+  "Creates a click action linking to an entity having `entity-type` with ID `entity-id`, in the normalized form.
+  `parameter-mapping` is an optional argument."
+  {:added "0.40.0"}
+  [entity-type entity-id & [parameter-mapping]]
+  (cond-> {::click-behavior-type ::link
+           ::link-type           entity-type
+           ::link-target-id      entity-id}
+          (some? parameter-mapping) (assoc ::parameter-mapping parameter-mapping)))
+
+(defn with-click-action
+  "Creates a click action from a given `from-field-id` Field identifier to the given `to-entity-type` having ID
+  `to-entity-id`, and adds it to the given `settings`. This happens in the normalized form, and hence this should be
+  passed the output of another fn (including, currently, `visualization-settings`). If the given `from-field-id`
+  already has a click action, it will be replaced."
+  {:added "0.40.0"}
+  [settings col-key action]
+  (-> settings
+      with-col-settings
+      (update-in [::column-settings] #(assoc % col-key {::click-behavior action}))))
```

Comment:
You don't need to use a function here. You can just do `(update-in ks assoc col-key ...)`

Reply:
Fixed.